opencti-platform/opencti-graphql/src/parser/bundle-creator.ts

```diff
@@ -2,6 +2,42 @@ import { v4 as uuidv4 } from 'uuid';
 import * as R from 'ramda';
 import type { StixBundle, StixObject } from '../types/stix-common';
 import { STIX_SPEC_VERSION } from '../database/stix';
+import { STIX_EXT_OCTI } from '../types/stix-extensions';
+
+/**
+ * Check if bundle object can be added to the current bundle or if a new bundle is required to use upsert feature.
+ * If there is the same stix id but with different content => it need to be in another bundle because
+ * Same stix id on the one bundle are removed from processing during worker split process (worker see them as duplicate).
+ * @param objectsToAdd
+ * @param bundles
+ */
+export const canAddObjectToBundle = (objectsToAdd: StixObject[], bundles: StixObject[]): boolean => {
+  const canAdd = true;
+  for (let i = 0; i < objectsToAdd.length; i += 1) {
+    const currentToCheck = objectsToAdd[i];
+    const existingObjectWithDifferentContent = bundles.find((item) => {
+      if (item.id === currentToCheck.id && item.type === currentToCheck.type) {
+        const itemClone = structuredClone(item);
+        if (itemClone.extensions) {
+          itemClone.extensions[STIX_EXT_OCTI].converter_csv = undefined;
+        }
+
+        const currentToCheckClone = structuredClone(currentToCheck);
```

Comment:
A small perf improvement could be to build the currentToCheckClone outside of the bundles.find(), to only clone it one time instead of on every bundle item